src/App.js

```diff
@@ -1,16 +1,85 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
 
 const App = () => {
+  const[entriesData, setEntriesData] = useState(chatMessages);
+
+  const likeEntry = (id) => {
+    setEntriesData((prev)=>{
```

Comment:
👍 The next state of `messages` depends on the previous state, so using the callback style of the `setState` method is a good call.